File: OrangeJuice.Server.Core/Configuration/JsonAwsOptionsConverter.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace OrangeJuice.Server.Configuration
{
	public sealed class JsonAwsOptionsConverter : IConverter<string, AwsOptions>
	{
		public AwsOptions Convert(string value)
		{
			return JObject.Parse(value)
						  .ToObject<AwsOptions>();
		}

		public string ConvertBack(AwsOptions value)
		{
			return JObject.FromObject(value)
						  .ToString(Formatting.None);
		}
	}
}

File: OrangeJuice.Server.Core.FSharp/Configuration/JsonAwsOptionsConverter.fs
namespace OrangeJuice.Server.FSharp.Configuration

open Newtonsoft.Json
open Newtonsoft.Json.Linq

open OrangeJuice.Server
open OrangeJuice.Server.Configuration

type JsonAwsOptionsConverter() =
    interface IConverter<string, AwsOptions> with
        member this.Convert(value : string) : AwsOptions =
            JObject.Parse(value).ToObject<AwsOptions>()

        member this.ConvertBack(value : AwsOptions) : string =
            JObject.FromObject(value).ToString(Formatting.None)
